src/api.c

```diff
@@ -6118,3 +6119,65 @@ const struct cgroup_library_version *cgroup_version(void)
 {
 	return &library_version;
 }
+
+/**
+ * Finds the current cgroup setup mode (legacy/unified/hybrid).
+ * Returns unknown of failure and setup mode on success.
+ */
+enum cg_setup_mode_t cgroup_setup_mode(void)
+{
+/* FS codes refered from include/magic.h */
+#define CGROUP2_SUPER_MAGIC     0x63677270
+#define TMPFS_MAGIC             0x01021994
+
+	const char *cgrp_sysfs_unified = "/sys/fs/cgroup/unified/";
```

Comment:
We have some users that don't use systemd.  Should we check for the existence of `/sys/fs/cgroup/`.  If it exists, use this shortcut.  Otherwise, look through the cg_mount_table.

Reply:
Or is this a "systemd-only" interface?  If `/sys/fs/cgroup` doesn't exist, should we just return an error?

Reply:
>  We have some users that don't use systemd. Should we check for the existence of /sys/fs/cgroup/. If it exists, use this shortcut. Otherwise, look through the cg_mount_table. Or is this a "systemd-only" interface? If /sys/fs/cgroup doesn't exist, should we just return an error?

I was thinking of the API from the view from systemd only. Agreed, I will re-think and come up with a better approach based on your suggestion of using the `cg_mount_table`.

Reply:
> Just beware the [directory in sysfs exists](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/kernel/cgroup/cgroup.c?id=833477fce7a14d43ae4c07f8ddc32fa5119471a2#n6000) regardless of systemd. What is different whether/what is mounted at that path. Mountinfo looks a more robust way in general landscape.

@Werkov Thank you for the review and testing of the patchset.  Yeah, the kernel creates the `/sys/fs/cgroup` during `cgroup_init()`. I will try to re-implement it using `cg_mount_table` that already captures the details about the mount points and also make it more robust by detecting all allowed cgroup setup combinations, those get influenced by `cgroup` kernel parameters.